engines/basic_resources/app/collections/basic_resources/memberships_collection.rb

```diff
@@ -0,0 +1,30 @@
+module BasicResources
+  # Fetches the list of memberships that belong to the user, in the given group
+  # if specified. All the memberships that belong to the user are returned,
+  # otherwise.
+  class MembershipsCollection
+
+    # Constructor
+    #
+    # @param user [User]
+    # @param group [Group]
+    def initialize(user, group)
+      @user = user
+      @group = group
+    end
+
+    # Returns the memberships of the user either scoped by the given group or
+    # all of them
+    #
+    # @return [ActiveRecord::Relation<Membership>]
+    def build
+      relation = Membership.where(user_id: user.id)
+      relation = relation.where(basic_resource_group_id: group.id) if group
```

Comment:
Actually, if `group_id` is not `nil` we need all the `Membership`s related to that group. Not `AND user_id = user.id`.

Scoped by `user_id` OR `basic_resource_group_id`